modules/opensearch/src/OpenSearchTingObject.php

```diff
@@ -89,15 +137,109 @@ public function setOwnerId($owner_id) {
     $this->ownerId = $owner_id;
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function getClassification() {
+    // Get the first classification.
+    $classification = $this->firstEntry($this->getRecordEntry('dc:subject', 'dkdcplus:DK5'));
+
+    if (empty($classification)) {
+      return FALSE;
+    }
+
+    // Ignore the shorthand for "Skønlitteratur".
+    return $classification === 'sk' ? '' : $classification;
+  }
 
   /**
-   * OpenSearchObject constructor.
-   *
-   * @param TingClientObject $open_search_object
-   *   The Open Search result this object wraps.
+   * {@inheritdoc}
    */
-  public function __construct($open_search_object) {
-    $this->openSearchObject = $open_search_object;
+  public function getCreators($format = TingObjectInterface::NAME_FORMAT_DEFAULT) {
+    // Get all authors, filter away instances of the field that is only used for
+    // search internals (eg, a sort-field).
+    if ($format == TingObjectInterface::NAME_FORMAT_SURNAME_FIRST) {
+      $creators = $this->getRecordEntry('dc:creator', 'oss:sort');
+    }
+    else {
+      // Filter away fields that should only be used for internals.
+      $search_only_fields = ['oss:sort'];
+      $creators = $this->filterRecordsExclude($this->getRecordLevel('dc:creator'), $search_only_fields);
+      // Each record entry is a single-element array - map it to a array of
+      // strings by picking the first element from each.
+      $creators = $this->recordsFlatten($creators);
+    }
+
+    return empty($creators) ? [] : $creators;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getSubjects() {
+    // Filter away fields that should only be used for internals.
+    $search_only_fields = ['dkdcplus:genre', 'dkdcplus:DK5', 'dkdcplus:DK5-Text', 'dkdcplus:DBCO', 'dkdcplus:DBCN'];
+    $subjects = $this->filterRecordsExclude($this->getRecordLevel('dc:subject'), $search_only_fields);
+    return $this->recordsFlatten($subjects);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getLanguage() {
+    return $this->firstEntry($this->getRecordEntry('dc:language'));
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function isOnline() {
+    // Check if the material has its own URI - if so we assume it is online.
+    return !empty($this->openSearchObject->record['dc:identifier']['dcterms:URI']);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getOnlineUrl() {
+    $url = '';
+    // Try to find the online url from relation data, which requires us to get
+    // relations. First check if relations are set; if not do another request
+    // to get relations.
+    if (!isset($this->openSearchObject->relationsData)) {
+      $ting_client_object = ting_get_object($this->getId(), FALSE, TRUE);
+      if (isset($ting_client_object->relationsData)) {
+        $this->openSearchObject->relationsData = $ting_client_object->relationsData;
+      }
+    }
```

Comment:
This seems to be duplicated from `getRelations()`. Could that be avoided?

Reply:
:+1: